### PR TITLE
Sighash to message

### PR DESCRIPTION
**Pull request by luckysori**
This is for https://github.com/comit-network/hermes/pull/16#discussion_r703115601.

What do you think? I tried to just use the trait you suggested, but there is a mismatch between the types because we're using `secp256k1` and `secp256k1-zkp`. There might be a better solution than what I've done though.

### CODE REVIEW

**Comment by thomaseizinger** [comment]
If we have a type mismatch then we need to bump the version of -zkp and release. Or where is the mismatch?
I assume bitcoin depends on a newer version of secp than -zkp does?

File: cfd_protocol/src/lib.rs [hunk 1]
```diff
@@ -875,6 +868,19 @@ impl TransactionExt for bitcoin::Transaction {
     }
 }
 
+trait SigHashExt {
+    fn as_message(&self) -> secp256k1_zkp::Message;
+}
+
+impl SigHashExt for SigHash {
+    fn as_message(&self) -> secp256k1_zkp::Message {
```
Comment (thomaseizinger) [inline cfd_protocol/src/lib.rs]: Should be `to_message` I think.

**Review by thomaseizinger — APPROVED** [review]
An extension trait is a good workaround until we can pass it along natively.

**Comment by luckysori** [comment]
> If we have a type mismatch then we need to bump the version of -zkp and release. Or where is the mismatch?
> I assume bitcoin depends on a newer version of secp than -zkp does?

I haven't managed to get to the bottom of the dependency mismatch problem, but I think you're on the right track. I'll merge this and record and issue.

**Comment by thomaseizinger** [comment]
> > If we have a type mismatch then we need to bump the version of -zkp and release. Or where is the mismatch?
> > I assume bitcoin depends on a newer version of secp than -zkp does?
> 
> I haven't managed to get to the bottom of the dependency mismatch problem, but I think you're on the right track. I'll merge this and record and issue.

I can't find any version mismatch in the latest `bitcoin` version. Weird ...

**Comment by luckysori** [comment]
> I can't find any version mismatch in the latest `bitcoin` version. Weird ...

I was just getting a headache trying to figure it out lol. Maybe we can take a look together or you can give it a go yourself.

**Pull request merged**